shark_turbine/kernel/lang/global_symbols.py

```diff
@@ -0,0 +1,21 @@
+from .._support.indexing import index_symbol
+
+# Global symbols used throughout the code.
+
+# Address spaces.
+GLOBAL_ADDRESS_SPACE = index_symbol("$GLOBAL_ADDRESS_SPACE")
+SHARED_ADDRESS_SPACE = index_symbol("$SHARED_ADDRESS_SPACE")
+
+# Distribution symbols.
+WORKGROUP_0 = index_symbol("$WG0")
+WORKGROUP_1 = index_symbol("$WG1")
+WORKGROUP_2 = index_symbol("$WG2")
+
+THREAD_0 = index_symbol("$T0")
+THREAD_1 = index_symbol("$T1")
+THREAD_2 = index_symbol("$T2")
+
+# MMA symbols
+MMA_LHS = index_symbol("$MMA_LHS")
```

Comment:
Is there some intuition to why do we change these but not `BLOCK_M`?

Reply:
Good question. BLOCK_M is a user-defined parameter. The user is free to choose any symbol for the tile size. But the symbols in this file are used in the codegen pipeline. (Think of them as reserved keywords in a language) and that's why we distinguish them.

Reply:
Makes sense, thanks!